src/modules/Elsa.Studio.Workflows/Components/WorkflowDefinitionEditor/Components/WorkflowDefinitionVersionViewer.razor.cs

```diff
@@ -67,34 +66,35 @@ private RadzenSplitterPane ActivityPropertiesPane
     /// <inheritdoc />
     protected override async Task OnInitializedAsync()
     {
+        _workflowDefinition = WorkflowDefinition;
         await ActivityRegistry.EnsureLoadedAsync();
 
-        if (WorkflowDefinition?.Root == null)
+        if (_workflowDefinition?.Root == null)
             return;
 
-        SelectActivity(WorkflowDefinition.Root);
+        SelectActivity(_workflowDefinition.Root);
     }
 
     /// <inheritdoc />
     protected override async Task OnParametersSetAsync()
     {
-        if (WorkflowDefinition?.Root == null)
+        if (WorkflowDefinition == _workflowDefinition)
+            return;
+     
+        _workflowDefinition = WorkflowDefinition;
+        
+        if (_workflowDefinition?.Root == null)
             return;
 
         if (_diagramDesigner != null)
-            await _diagramDesigner.LoadActivityAsync(WorkflowDefinition.Root);
+            await _diagramDesigner.LoadActivityAsync(_workflowDefinition.Root);
 
-        SelectActivity(WorkflowDefinition.Root);
+        SelectActivity(_workflowDefinition.Root);
     }
 
     /// <inheritdoc />
     protected override async Task OnAfterRenderAsync(bool firstRender)
```

Comment:
Do we still need to overwrite the method if it is empty?

Reply:
Not at all.